### PR TITLE
Configure retries in GeneralUtils and add tests

### PR DESCRIPTION
**Pull request by lpatalas**
- Add tests for `IsFeedPublicAsync` and `CompareLocalPackageToFeedPackage`
- Configure retry for these functions to `MaxAttempts` = 5 and `DelayBase` = 5

Implements: https://github.com/dotnet/core-eng/issues/10431

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/GeneralUtils.cs [hunk 1]
```diff
@@ -117,6 +123,39 @@ public static async Task<bool> CompareStreamsAsync(Stream localFileStream, Strea
...
         /// <param name="client"></param>
+        /// <param name="log"></param>
+        /// <returns></returns>
+        /// <remarks>
+        ///     Open a stream to the local file and an http request to the package. There are a couple possibilities:
+        ///     - The returned headers includes a content MD5 header, in which case we can
+        ///       hash the local file and just compare those.
+        ///     - No content MD5 hash, and the streams must be compared in blocks. This is a bit trickier to do efficiently,
+        ///       since we do not necessarily want to read all bytes if we can help it. Thus, we should compare in blocks.  However,
+        ///       the streams make no gaurantee that they will return a full block each time when read operations are performed, so we
```
Comment (MattGal) [inline src/Microsoft.DotNet.Build.Tasks.Feed/src/common/GeneralUtils.cs:134]: nits: "returned headers include" (no s) and "guarantee"

File: src/Common/IRetryHandler.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
+using System;
```
Comment (riarenas) [inline src/Common/IRetryHandler.cs:1]: [nit] License header

File: src/Microsoft.DotNet.Build.Tasks.Feed.Tests/TestDoubles/MockRetryHandler.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
+using System;
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed.Tests/TestDoubles/MockRetryHandler.cs:1]: [nit] license header

File: src/Microsoft.DotNet.Build.Tasks.Feed.Tests/TestInputs.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+using System.IO;
```
Comment (riarenas) [inline src/Microsoft.DotNet.Build.Tasks.Feed.Tests/TestInputs.cs:1]: [nit, sorry] license header

File: src/Microsoft.DotNet.Build.Tasks.Feed/src/common/GeneralUtils.cs (hunk 1 above)
Comment (lpatalas) [reply src/Microsoft.DotNet.Build.Tasks.Feed/src/common/GeneralUtils.cs]: Fixed

File: src/Common/IRetryHandler.cs (hunk 2 above)
Comment (lpatalas) [reply src/Common/IRetryHandler.cs:5]: Good catch, I forgot about them. Fixed now.

**Comment by riarenas** [comment]
looks like existing tests and / or task code need to be updated to account for the changes in how you wire the retries?

**Comment by lpatalas** [comment]
I think it's just a silly mistake of concatenating paths manually instead of using `Path.Combine`. I'm checking it now.

**Pull request merged**